### ISSUE TITLE
Unable to use low-privilege accounts in containers

### ISSUE THREAD

**Issue by Randark-JMT** [ktock/container2wasm#102]
I used the following Dockerfile to build the test image

```dockerfile
FROM ubuntu:22.04

RUN useradd -m test && \
    echo "test:123" | chpasswd

USER test
```

Once the image is compiled, convert it to a wasm file and start the service, the following error is reported

![image](https://github.com/ktock/container2wasm/assets/56223739/ea83b24a-ece9-4d77-b0ac-3c9e3bdc0eeb)

```plaintext
/bin/bash: error while loading shared libraries: libtinfo.so.6: cannot open shared object file: Permission denied
```

**Comment by Randark-JMT** [comment]
After switching to the root account for testing, you can find that this problem can be temporarily circumvented by executing the following command to change the file permissions

![image](https://github.com/ktock/container2wasm/assets/56223739/a46bfc5d-742c-4e96-8110-8e28daf0a545)

**Comment by Randark-JMT** [comment]
I've tested the above images directly with the original docker and they all run fine, I suspect the problem is in the Bochs layer, but that's just a guess.

**Comment by ktock** [comment]
@Randark-JMT I believe this has been fixed by v0.4.0. Could you try this with v0.4.0 or later?

**Comment by Randark-JMT** [comment]
Great, it's true that this problem exists on 0.3.0, but I've tested it on 0.4.0 and there's no problem!

Thank you very much for your reply!
